scipy/signal/_peak_finding.py

```diff
@@ -257,24 +257,39 @@ def peak_prominences(vector, peaks, wlen=None):
         Indices of peaks in `vector`.
     wlen : int, optional
         A window lenght in samples that limits the search for the lowest
-        contour line two a symmetric interval around the evaluated peak. If
-        not given the entire vector is used. Use this parameter to speed up
-        the calculation significantly for large vectors.
+        contour line two a symmetric interval around the evaluated peak. If not
+        given the entire vector is used. Use this parameter to speed up the
+        calculation significantly for large vectors.
```

Comment:
When is it large? Why don't we always use it then ? and so on. I get what you are saying but it's not helping a standard user.

Reply:
This is vague on purpose because I can't exactly pinpoint an array size after which there is a significant speedup. Maybe I could try to reword or expand on this in the notes? Something like:

"Searching for the peak's bases can be slow for large vector because the full vector needs to be evaluated for each peak. This evaluation area can be limited with the parameter `wlen` which restricts the algorithm to a window around the current peak and can shorten the calculation time significantly if `wlen` is significantly smaller than `vector`. However this may stop the algorithm from finding the true global contour line if the peak's bases are outside this window. Instead a higher contour line is found within the restricted window leading to a smaller calculated prominence. In practice this is only relevant for the largest set of peaks in `vector`. This behavior may even be used intentionally to calculate "local" prominences.

Reply:
+1 for saying `see Notes` somewhere here and adding this note below, I agree it depends on a lot of factors so we can't be specific

Reply:
yes see notes would be indeed better